suite/suite.go

```diff
@@ -84,21 +121,13 @@ func Run(t *testing.T, suite TestingSuite) {
 				F: func(t *testing.T) {
 					parentT := suite.T()
 					suite.SetT(t)
-					if setupTestSuite, ok := suite.(SetupTestSuite); ok {
-						setupTestSuite.SetupTest()
-					}
-					if beforeTestSuite, ok := suite.(BeforeTest); ok {
-						beforeTestSuite.BeforeTest(methodFinder.Elem().Name(), method.Name)
-					}
+					beforeHandlers(suite, suiteName, method.Name)
+
 					defer func() {
-						if afterTestSuite, ok := suite.(AfterTest); ok {
-							afterTestSuite.AfterTest(methodFinder.Elem().Name(), method.Name)
-						}
-						if tearDownTestSuite, ok := suite.(TearDownTestSuite); ok {
-							tearDownTestSuite.TearDownTest()
-						}
+						afterHandlers(suite, suiteName, method.Name)
 						suite.SetT(parentT)
 					}()
+
 					method.Func.Call([]reflect.Value{reflect.ValueOf(suite)})
```

Comment:
to measure the time of the function. You should to add b.RestartTim and then b.startTime and after the calling b.StopTime. Otherwise the result would be wrong